engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/compensation/CompensationEventExecutionTest.java

```diff
@@ -899,6 +902,152 @@ public void shouldNotCreateSubprocessSubscriptionWithoutChildSubscription() {
                 ProcessInstanceIntent.ELEMENT_COMPLETED));
   }
 
+  @Test
+  public void shouldInvokeCompensationHandlerTheSameAmountAsExecuted() {
+    // given
+    final var process =
+        createModelFromClasspathResource("/compensation/compensation-multi-instance.bpmn");
+    ENGINE.deployment().withXmlResource(process).deploy();
+    final long processInstanceKey = ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+
+    ENGINE
+        .jobs()
+        .withType(SERVICE_TASK_TYPE_COMPENSABLE_ACTIVITY)
+        .activate()
+        .getValue()
+        .getJobKeys()
+        .forEach(jobKey -> ENGINE.job().withKey(jobKey).complete());
+
+    RecordingExporter.jobRecords(JobIntent.CREATED)
+        .withProcessInstanceKey(processInstanceKey)
+        .withElementId("CompensationHandler")
+        .map(Record::getKey)
+        .forEach(jobKey -> ENGINE.job().withKey(jobKey).complete());
```

Comment:
:wrench: The test case waits here for 5 seconds because the job record stream is not limited.